app/controllers/application_controller.rb

```diff
@@ -4,13 +4,24 @@ class ApplicationController < ActionController::Base
 
   rescue_from ActionController::InvalidAuthenticityToken, with: :session_expired
   rescue_from ActionController::RoutingError, with: :render_not_found
+  rescue_from GetIntoTeachingApiClient::ApiError, with: :handle_api_error
 
   def raise_not_found
     raise ActionController::RoutingError, "Not Found"
   end
 
 private
 
+  def handle_api_error(error)
+    render_too_many_requests && return if error.code == 429
```

Comment:
Feels like it would be nicer to identify this as a 429 within the client, and raise a specific descriptive exception (eg `GetIntoTeachingApiClient::RateLimitExceededError`) in the client. Then you could rescue that particular exception here?

Reply:
I agree that would be a little cleaner, but IMO the fact we'd have to add to the monkey patching to achieve it outweighs the benefit it brings. I'd prefer to limit the amount of patching we're doing if possible, but if you feel strongly about it happy to discuss?

Reply:
Ok, lets leave it for now.